### PR TITLE
Document tests and clean up

### PR DESCRIPTION
**Pull request by Nocccer**
With this pr i tried to discribe the frontend test enviroment.
This should help people understanding how the frontend test's working and to adapt them on changes.

- [x] document testTypes
- [x] document electron mock
- [x] added doc folder with some information about frontend test enviroment
- [x] clean up

### CODE REVIEW

**Review by dragonDScript — APPROVED** [review]

**Comment by flavioislima** [comment]
I'll take a look when I have time because I want to understand 😁

**Comment by Flonky** [comment]
I'll take a look at the documentation in a bit, don't immediately have time for it

**Comment by Nocccer** [comment]
@flavioislima @Flonky take your time. Doesn't rush to merge it.

File: src/test_helpers/mock/electron.ts [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
     .calledWith(stringAtIndex(0, 'egsSync')).mockResolvedValue(test_egssync_response.get())
-    .calledWith('getAlternativeWine').mockResolvedValue([test_wineinstallation.get()])
-    .calledWith(stringAtIndex(0, 'openMessageBox')).mockResolvedValue({response: test_openmessagebox_response.get()})
+    .calledWith('getAlternativeWine').mockResolvedValue(test_wineinstallations.get())
+    .calledWith(stringAtIndex(0, 'openMessageBox')).mockResolvedValue(
+      {response: test_openmessagebox_response.get()})
     .calledWith(stringAtIndex(0, 'syncSaves')).mockResolvedValue('success')
-    .calledWith(stringAtIndex(0, 'writeConfig')).mockResolvedValue({});
+    .calledWith(stringAtIndex(0, 'writeConfig')).mockResolvedValue({})
+    .calledWith(stringAtIndex(0, 'callTool')).mockResolvedValue({});;
```
Comment (Nocccer) [inline src/test_helpers/mock/electron.ts:58]:
```suggestion
    .calledWith(stringAtIndex(0, 'callTool')).mockResolvedValue({});
```

**Review by flavioislima — APPROVED** [review]

**Comment by flavioislima** [comment]
Amazing piece of work here @Nocccer 
Thanks for that.
I'll find time to improve it as well :)

**Pull request merged**